### PR TITLE
[DO-1974] Fix saving nacp id of related persons

### PR DESCRIPTION
**Pull request by EmmaSamoilova**

### CODE REVIEW

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -1010,46 +1020,37 @@ def find_city(self, address_data):
...
+                        else:
+                            related_person.nacp_id.append(related_person_nacp_id)
+                            related_person.save()
+                        # TODO: decide should we store new Pep that not spouse from relatives_data
+                        if to_person_relationship_type in SPOUSE_TYPES:
+                            declaration.spouse = related_person
+                            declaration.save()
+                except InvalidRelativeData as e:
+                    self.log_error(f'{e}')
+                    continue
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:1051]: you can catch this error only in is_same_full_name

File: business_register/management/commands/fetch_peps_nacp_id.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def __init__(self, *args, **kwargs):
         """)
         self.peps_without_nacp_id = {getattr(pep, 'source_id'): pep for pep in Pep.objects.filter(
             is_pep=True,
-            nacp_id__isnull=True
+            nacp_id=[]
```
Comment (roman-tiukh) [inline business_register/management/commands/fetch_peps_nacp_id.py:84]: __len=0?

File: business_register/migrations/0118_auto_20210611_0951.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        migrations.AddField(
+            model_name='historicalpep',
+            name='nacp_id',
+            field=django.contrib.postgres.fields.ArrayField(base_field=models.PositiveBigIntegerField(unique=True, verbose_name='id from NACP'), blank=True, default=list, help_text='id from the National agency on corruption prevention', size=None),
+        ),
+        migrations.AddField(
+            model_name='pep',
+            name='nacp_id',
+            field=django.contrib.postgres.fields.ArrayField(base_field=models.PositiveBigIntegerField(unique=True, verbose_name='id from NACP'), blank=True, default=list, help_text='id from the National agency on corruption prevention', size=None),
+        ),
```
Comment (roman-tiukh) [inline business_register/migrations/0118_auto_20210611_0951.py:31]: you can save all id:
1. add array-field `nacp_id1`
2. move id data from `nacp_id` to `nacp_id1`
3. remove field `nacp_id`
4. rename field `nacp_id1` to `nacp_id`

File: business_register/converter/declaration.py [hunk 4]
```diff
@@ -65,6 +65,17 @@ def __init__(self):
...
+    def find_person(self, pep_id):
+        pep = Pep.objects.filter(nacp_id__contains=[int(pep_id)])
+        if pep and len(pep) == 1:
+            pep = pep[0]
+        # TODO: decide what to do if more than one person has the same nacp_id
+        elif pep and len(pep) > 1:
+            self.log_error(f'More than one person has nacp_id = {pep_id}')
+        else:
+            pep = None
+        return pep
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:77]:
```suggestion
    def find_person(self, pep_id):
        peps = list(Pep.objects.filter(nacp_id__contains=[int(pep_id)]))
        if len(peps) == 1:
            return peps[0]
        elif len(peps) > 1:
            self.log_error(f'More than one person has nacp_id = {pep_id}')
        return None
```

File: business_register/converter/declaration.py [hunk 5]
```diff
@@ -1010,46 +1020,37 @@ def find_city(self, address_data):
...
-                            self.log_error(f'{e}')
-                            break
-                if spouse:
-                    declaration.spouse = spouse
-                    declaration.save()
-                    break
+            related_person_links = RelatedPersonsLink.objects.filter(
+                         from_person=pep,
+                         to_person_relationship_type=to_person_relationship_type
+            )
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:1031]:
```suggestion
            related_person_links = RelatedPersonsLink.objects.filter(
                from_person=pep,
                to_person_relationship_type=to_person_relationship_type
            )
```

File: business_register/management/commands/fetch_peps_nacp_id.py [hunk 6]
```diff
@@ -56,9 +56,9 @@ def is_same_full_name(relative_data, pep):
     if not middle_name:
         middle_name = ''
     return (
-            pep.last_name.capitalize() == last_name
-            and pep.first_name.capitalize() == first_name
-            and pep.middle_name.capitalize() == middle_name
+            pep.last_name.capitalize() == last_name.capitalize().strip()
+            and pep.first_name.capitalize() == first_name.capitalize().strip()
+            and pep.middle_name.capitalize() == middle_name.capitalize().strip()
```
Comment (roman-tiukh) [inline business_register/management/commands/fetch_peps_nacp_id.py:61]:
```suggestion
            pep.last_name.lower() == last_name.lower().strip()
            and pep.first_name.lower() == first_name.lower().strip()
            and pep.middle_name.lower() == middle_name.lower().strip()
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/declaration.py (hunk 1 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/management/commands/fetch_peps_nacp_id.py (hunk 2 above)
Comment (EmmaSamoilova) [reply business_register/management/commands/fetch_peps_nacp_id.py]: done

File: business_register/converter/declaration.py (hunk 4 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/management/commands/fetch_peps_nacp_id.py (hunk 6 above)
Comment (EmmaSamoilova) [reply business_register/management/commands/fetch_peps_nacp_id.py]: done

File: business_register/migrations/0118_auto_20210611_0951.py (hunk 3 above)
Comment (EmmaSamoilova) [reply business_register/migrations/0118_auto_20210611_0951.py]: done, but I'm not sure about the SQL query

File: business_register/converter/declaration.py (hunk 5 above)
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/migrations/0118_auto_20210614_0723.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        migrations.RunSQL(
+            sql=f'''
+                UPDATE business_register_pep
+                SET nacp_id1 = array
+                  (SELECT nacp_id
+                   FROM
+                     (SELECT ROW(nacp_id)
+                      FROM business_register_pep
+                      LIMIT 1) AS pep_row)
+                WHERE nacp_id IS NOT NULL;
```
Comment (roman-tiukh) [inline business_register/migrations/0118_auto_20210614_0723.py:37]: is this sql works??

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/migrations/0118_auto_20210614_0723.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                base_field=models.PositiveBigIntegerField(unique=True, verbose_name='id from NACP'), blank=True,
+                default=list, help_text='id from the National agency on corruption prevention', size=None),
+        ),
+        migrations.AddField(
+            model_name='pep',
+            name='nacp_id1',
+            field=django.contrib.postgres.fields.ArrayField(
+                base_field=models.PositiveBigIntegerField(unique=True, verbose_name='id from NACP'), blank=True,
+                default=list, help_text='id from the National agency on corruption prevention', size=None),
+        ),
```
[the remark this answers is not in the document]
Comment (EmmaSamoilova) [reply business_register/migrations/0118_auto_20210614_0723.py]: 'historicalpep' and 'pep'

File: business_register/migrations/0118_auto_20210614_0723.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        migrations.RenameField(
+            model_name='historicalpep',
+            old_name='nacp_id1',
+            new_name='nacp_id',
+        ),
+        migrations.RenameField(
+            model_name='pep',
+            old_name='nacp_id1',
+            new_name='nacp_id',
+        )
```
[the remark this answers is not in the document]
Comment (EmmaSamoilova) [reply business_register/migrations/0118_auto_20210614_0723.py]: 'historicalpep' and 'pep'

File: business_register/migrations/0118_auto_20210614_0723.py (hunk 7 above)
Comment (EmmaSamoilova) [reply business_register/migrations/0118_auto_20210614_0723.py]: fix

**Pull request closed without merging**